magento-research/pwa-studio
# Passes email, firstname, and lastname to checkout receipt properly

## Description


After a successful guest checkout, users are prompted to create an account:

![Screen Shot 2019-04-22 at 3 10 43 PM](https://user-images.githubusercontent.com/13182778/56525623-dd686800-6510-11e9-8e70-74be42acf565.png)

The design is that their first name, last name, and email would be pre-set for them on the "Create Account" page, with that information being taken from the previous order's details:

![Screen Shot 2019-04-22 at 3 14 23 PM](https://user-images.githubusercontent.com/13182778/56526040-6aabbc80-6511-11e9-8310-8fbdefffda41.png)

Unfortunately, after being sent to the "Create Account" page, these fields were filled with `null`s:

![image](https://user-images.githubusercontent.com/13182778/56525258-59ae7b80-6510-11e9-9b96-2766b8444c79.png)

This PR fixes up some internal wiring so that those fields are once again pre-populated with the correct data (not `null`).

## Related Issue





Closes #830 .

## Verification Steps


Perform the following steps twice: Once for billing address same as shipping address, and once with billing address different than shipping address:

![Screen Shot 2019-04-22 at 3 10 28 PM](https://user-images.githubusercontent.com/13182778/56526170-975fd400-6511-11e9-93e3-eee0c2616c07.png)

1. Complete the checkout process as a guest user and submit your order, making a note of the values you enter for shipping address first name, last name, and email fields.
1. On the receipt screen, click to `Create an Account`
1. See that the first name, last name, and email fields are populated with the data entered in step 1.

📝 We don't have to test this for authenticated users because the "Create Account" button does not exist for them on the checkout receipt (they already have an account):

![Screen Shot 2019-04-24 at 1 23 04 PM](https://user-images.githubusercontent.com/13182778/56683929-3dd3e280-6694-11e9-91a8-e3fd3b1839cb.png)

## How Have YOU Tested this?




1. The steps above
1. `yarn test`

## Screenshots / Screen Captures (if appropriate):

## Proposed Labels for Change Type/Package




## Checklist:
